galpy/orbit/Orbits.py

```diff
@@ -1483,13 +1547,41 @@ def integrate(
             raise ValueError(
                 "dt input (integrator stepsize) for Orbit.integrate must be an integer divisor of the output stepsize"
             )
-        # Delete attributes for interpolation and rperi etc. determination
-        if hasattr(self, "_orbInterp"):
-            delattr(self, "_orbInterp")
+
+        # Prepare potential for comparison
         if self.dim() == 2:
             thispot = toPlanarPotential(pot)
         else:
             thispot = pot
+
+        # Check if we should continue from a previous integration
+        should_continue, is_forward, pot_changed = self._should_continue_integration(
+            numpy.array(t), thispot
+        )
+
+        # Warn if continuing with a different potential
+        if should_continue and pot_changed:
+            warnings.warn(
+                "Continuing orbit integration with a different potential than the previous integration; this may lead to unphysical results",
+                galpyWarning,
+            )
+
+        # Store old orbit data if continuing
+        if should_continue:
+            old_t = self.t.copy()
+            old_orbit = self.orbit.copy()
+            # Update initial conditions to be the final state of previous integration
+            if is_forward:
+                # For forward continuation, start from the last state
+                self.vxvv = self.orbit[:, -1, :].copy()
+            else:
+                # For backward continuation, start from the first state
+                self.vxvv = self.orbit[:, 0, :].copy()
```

Comment:
Once the continued integration is finished, we want to go back to having vxvv be the original initial condition, so it needs to be cached and restored. This is to keep vxvv consistent between different ways of integration (that is, it should always be the original initial condition, although briefly changing it internally is fine)